Support multiple generic backends

aedades opened the pull request:
- Displays secrets under `/secret/` by default
- Can select other _generic_ type backends by clicking on `ROOT`

aedades committed:
Add ability to list secret backends

aedades committed:
Fix endpoint

Commit:
Listen for /listsecretbackends

Commit:
Grab secret backends

Commit:
Set default namespace to /secret/

Commit:
Merge branch 'master' into support-multiple-backends

Commit:
drop /secret - namespace included in secret path

Commit:
List 'generic' type backends or secrets depending on whether user clicks ROOT

Commit:
Display secrets under /secret/ by default

djenriquez commented:
What you think @Lucretius?

djenriquez commented:
Ok, definitely supports multiple backends, but I think theres some tweaking we can do here to provide a better experience. I'll play around with it a bit later tonight 👍

djenriquez committed:
Fix indentation

djenriquez committed:
Disable add button on generic secret list

- Rename variable for clarity
- Disable button when selecting generic backend. Users should not have
access to add keys from this location since that would mean enabling
other generic backends.

djenriquez committed:
Fixing conflict

djenriquez committed:
Merge branch 'master' into support-multiple-backends
